automation_tools/utils.py

```diff
@@ -79,3 +82,37 @@ def update_packages(*args, **kwargs):
         quiet=kwargs.get('quiet', False),
         warn_only=kwargs.get('warn_only', False),
     )
+
+
+def get_discovery_image():
+    """ Task for getting unattended foreman-discovery ISO image
+    :return: foreman-discovery-image iso under /var/lib/libvirt/images/
+    """
+    url = os.environ.get('BASE_URL') + '/Packages/'
+    try:
+        from BeautifulSoup import BeautifulSoup
+    except Exception:
+        subprocess.call(['pip', 'install', 'BeautifulSoup'])
```

Comment:
Instead, you add this is requirements, which will install BeautifulSoap for sure. And then Remove try block

Reply:
@jyejare yes I thought of that. But other side is why to add extra requirement to whole project (causing to install it everytime thought it is not required) for just one task. What do you think?

Reply:
True ! +1

Reply:
AFAIK, imports into a local scope, such as inside a function definition should be avoided, people prefer all the imports to happen at the top so that anyone can quickly know what are the dependencies.

Reply:
@ntkathole , would request an update to the PR and move the import statement for `BeautifulSoap` to the top of the
module.

Reply:
Also, the `BeautifulSoup` as a dependency should go inside `install_requires` under setup.py

https://github.com/SatelliteQE/automation-tools/blob/master/setup.py#L31

Reply:
As to why it should be avoided is a good question, but with my experience, I say it is not a good practice to handle installation of the modules inside the code. 

Maybe this link helps, https://stackoverflow.com/questions/6120902/how-do-i-automatically-install-missing-python-modules?noredirect=1